test/solver/batch_cg_kernels.cpp

```diff
@@ -179,7 +179,8 @@ TEST_F(BatchCg, CanSolveLargeBatchSizeHpdSystem)
     auto mat =
         gko::share(gko::test::generate_diag_dominant_batch_matrix<const Mtx>(
             exec, num_batch_items, num_rows, true));
-    auto linear_system = setup_linsys_and_solver(mat, num_rhs, tol, max_iters);
+    auto linear_system =
+        setup_linsys_and_solver(mat, num_rhs, tol / 100, max_iters);
```

Comment:
Stopping by residual norm but checking the true error is still weird to me. the scale is 50000, which is a little high to me.
you also check the residual norm, so I do not hold this pr by this question now

Reply:
Do you suggest I dont check with the true solution at all, because I am definitely having issues with DPCPP with the tolerance. I also agree that 500 is too high.

Reply:
Yes, but the tol needs to be lower than current setup.
If it is the issue only in dpcpp, I think we need to be more careful on this.
For example, using the same n_shared settings, subgroup_size, group_size, and maybe using the same impl of reduction not from reduce_by_group on sycl and cuda side. If they still give quite different result, I think there are something wrong in the sync.